### PR TITLE
Rename `id` fields in the input or output which were dropped earlier

### LINKED ISSUE #508: GroupKey Id missing from cloudidentity Group
### What happened?

The cloudidentity.Group response is missing the groupKey.id

### Steps to reproduce

1. Create a service account with a `Google Groups Admin Role` in Workspace, I followed: https://medium.com/google-cloud/create-google-groups-via-terraform-de99524f92e0

2. Create a new cloudIdentity group and see that it is missing the `.groupKey.id`:

```typescript
const group = new cloudidentity.v1.Group("foobar", {
  groupKey: {id: "foobar@example.com"},
  initialGroupConfig: "EMPTY",
  labels: {
    "cloudidentity.googleapis.com/groups.discussion_forum": "",
  },
  parent: `customers/${customerId}`
})
```

3. See that `group.groupKey.id` is `undefined` and does not compile.

### Expected Behavior

`group.groupKey.id` is an `Output(foobar@exmaple.com)`

### Actual Behavior

it's `undefined`

### Versions used

CLI
Version      3.33.2
Go Version   go1.17.10
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host
OS       ubuntu
Version  20.04
Arch     x86_64

This project is written in nodejs (/home/kaoru/.asdf/shims/node v16.14.2)

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by viveklak**
Previously all `id` fields in the inputs or outputs were quietly dropped. Here we rename them to avoid collision with the `id` field injected by Pulumi.

Fixes #508
Although the above issue only mentions not being able to address `id` fields in outputs, I found that resource inputs with the name `id` were being silently dropped in the engine.

This supersedes #521

### CODE REVIEW

**Comment by viveklak** [comment]
> * google-native:apigee/v1:Apicategory

This is because previously the schema generation was failing for this resource with:
```
Failed to build ID params for resource google-native:apigee/v1:Apicategory: property "apicategoriesId" not found in "v1/organizations/{organizationsId}/sites/{sitesId}/apicategories/{apicategoriesId}"
```

File: Makefile [hunk 1]
```diff
@@ -26,7 +26,7 @@ local_generate:: bin/pulumi-java-gen
 generate_schema:: bin/pulumi-java-gen
 	echo "Generating Pulumi schema..."
 	$(WORKING_DIR)/bin/$(CODEGEN) schema ${VERSION}
-	$(WORKING_DIR)/bin/$(JAVA_GEN) generate --schema $(WORKING_DIR)/provider/cmd/$(PROVIDER)/schema.json --out sdk/java --build gradle-nexus
+	#$(WORKING_DIR)/bin/$(JAVA_GEN) generate --schema $(WORKING_DIR)/provider/cmd/$(PROVIDER)/schema.json --out sdk/java --build gradle-nexus
```
Comment (lblackstone) [inline Makefile:29]: Still needed? Actually, it looks like the whole Makefile should be updated from `ci-mgmt`.

File: provider/pkg/gen/schema.go [hunk 2]
```diff
@@ -1008,6 +1014,43 @@ func (g *packageGenerator) setOperationsBaseURL(cloudOp *resources.CloudAPIOpera
...
+	props, exists := resourcePropBag.specs["id"]
+	if !exists {
+		return
+	}
+
+	tok := g.genToken(typeName)
+	ignoreCollision := false
+	if _, exists = resourcePropBag.specs[newId]; exists {
+		if tok == "google-native:baremetalsolution/v2:NfsShare" {
+			ignoreCollision = true
```
Comment (lblackstone) [inline provider/pkg/gen/schema.go:1031]: Might be worth a comment explaining why we want to ignore collisions here.

**Review by lblackstone — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (viveklak) [reply Makefile:29]: Doh - this leaked through from my dev iteration changes. Will remove this.

File: provider/pkg/gen/schema.go (hunk 2 above)
Comment (viveklak) [reply provider/pkg/gen/schema.go:1033]: Done

**Review by lblackstone — APPROVED** [review]

**Pull request closed without merging**